README.md

```diff
@@ -1,6 +1,306 @@
 # CNS Object Broker
-## For dynmically provisioning object storage compliant with an S3 interface.
+## Utilize Kubernetes Service-Catalog to dynamically provision CNS Object Storage.
 
 ## WARNING!
+The work in this project is a proof of concept and not intended production.
 
-The work in this project is a proof of concept and not intended for anything more than that.
+## Overview
+A core feature of the Kubernetes system is the ability to provision a diverse
+offering of block and file storage on demand.
+This project seeks to demonstrate
+that by using the Kubernetes-Incubator's Service-Catalog, it is now also possible
+to bring this dynamic provisioning to S3 object storage.
+This broker is designed
+to be used with [Gluster-Kubernetes](https://github.com/jarrpa/gluster-kubernetes).
+
+See our detailed [command flow diagram](docs/diagram/control-diag.md) for
+ an overview of the system.
+
+### What It Does
+This broker implements OpenServiceBroker methods for creating, connecting to, and destroying
+object buckets.
+- For each new Service Instance, a new, uniquely named bucket is created.
+- For each new Service Instance Credential, a Secret is generated with the
+coordinates and credentials of the Service Instance's associated bucket.
+- Deleting a Service Instance destroys the associated bucket.
+- Deleting a Service Instance Credential deletes the secret.
+Nothing is done on the broker side.
+This is because the broker does not perform any actions on Bind, so there is nothing to
+undo.
+
+### Limitations
+
+- Currently, the CNS Object Broker is dependent on GCE.  If run outside of a GCE environment, it will fail to start.  This is because the Broker detects the external IP of the node on which it is run.
```

Comment:
_...environment, it will fail to start~.~ ~This is~ because the Broker detects the external_